### PR TITLE
folioreader class implemented/bookid and port number can be customized

### PR DESCRIPTION
**Pull request by avez321**

### CODE REVIEW

File: folioreader/src/main/java/com/folioreader/ui/folio/activity/FolioActivity.java [hunk 1]
```diff
@@ -199,7 +205,8 @@ public void onClick(View v) {
 
     private void initBook(String mEpubFileName, int mEpubRawId, String mEpubFilePath, EpubSourceType mEpubSourceType) {
         try {
-            mEpubServer = EpubServerSingleton.getEpubServerInstance(Constants.PORT_NUMBER);
+            int portNumber = getIntent().getIntExtra(Config.INTENT_PORT, 8080);
```
Comment (GautamChibde) [inline folioreader/src/main/java/com/folioreader/ui/folio/activity/FolioActivity.java]: use Constants.PORT_NUMBER instead hard coded 8080

File: folioreader/src/main/java/com/folioreader/util/FolioReader.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import com.folioreader.ui.folio.activity.FolioActivity;
+
+/**
+ * Created by avez raj on 9/13/2017.
+ */
+
+public class FolioReader {
+    public static final String INTENT_BOOK_ID = "book_id";
+
+    public void openBook(String assetsUrl, int rawId, Context context) {
```
Comment (mahavir155) [inline folioreader/src/main/java/com/folioreader/util/FolioReader.java]: first parameter context

Comment (mahavir155) [reply folioreader/src/main/java/com/folioreader/util/FolioReader.java]: also create two methods:
openBook(Context context, String assetOrSdcardPath)
openBook(Context, int rawId)

File: folioreader/src/main/java/com/folioreader/util/FolioReader.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Intent intent = getIntentFromUrl(assetsUrl,rawId,context);
+        context.startActivity(intent);
+    }
+
+    public void openBook(String assetsUrl, int rawId ,Config config, int port, Context context) {
```
Comment (mahavir155) [inline folioreader/src/main/java/com/folioreader/util/FolioReader.java]: Same, 
1. context as first parameter
2. create following methods:
- openBook(Context context, String assetOrSdcardPath, Config config)
- openBook(Context context, int rawId, Config config)

- openBook(Context context, String assetOrSdcardPath, Config config, int port)
- openBook(Context context, int rawId, Config config, int port)

- openBook(Context context, String assetOrSdcardPath, Config config, int port, String bookId)
- openBook(Context context, int rawId, Config config, int port, String bookId)

File: folioreader/src/main/java/com/folioreader/util/FolioReader.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    public void openBook(String epubBookUrl,int rawId, Config config, String bookId, int port, Context context) {
+        Intent intent = getIntentFromUrl(epubBookUrl,rawId,context);
+        intent.putExtra(Config.INTENT_CONFIG,config);
+        intent.putExtra(INTENT_BOOK_ID, bookId);
+        intent.putExtra(Config.INTENT_PORT, port);
+        context.startActivity(intent);
+
+    }
+
+    private Intent getIntentFromUrl(String assetsUrl, int rawId,Context context) {
```
Comment (mahavir155) [inline folioreader/src/main/java/com/folioreader/util/FolioReader.java]: change method to following:
getIntentFromUrl(Context context, String assetOrSdcardUrl, int rawId)
also handle books from SD card. check if url is asset url, then use asset else SD card

File: folioreader/src/main/java/com/folioreader/ui/folio/activity/FolioActivity.java [hunk 5]
```diff
@@ -135,6 +138,8 @@ protected void onCreate(Bundle savedInstanceState) {
         if(!mConfig.isShowTts()){
             findViewById(R.id.btn_speaker).setVisibility(View.GONE);
         }
+
+        mBookId = getIntent().getStringExtra(FolioReader.INTENT_BOOK_ID);
```
Comment (mahavir155) [inline folioreader/src/main/java/com/folioreader/ui/folio/activity/FolioActivity.java]: check if bookId is null, then use hash code of book title

File: sample/src/main/java/comfolioreader/android/sample/HomeActivity.java [hunk 6]
```diff
@@ -50,7 +51,7 @@ public void onClick(View v) {
                 if (ContextCompat.checkSelfPermission(HomeActivity.this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
```
Comment (mahavir155) [inline sample/src/main/java/comfolioreader/android/sample/HomeActivity.java]: please move permission checking in FolioReaderActivity class

**Pull request merged**